R/util.R

```diff
@@ -67,6 +68,23 @@ select_assay <- function(obj) {
   NULL
 }
 
+#' Extract the counts matrix from the Assay
+#'
+#' @param assay A SeuratObject::Assay or SeuratObject::Assay5
+#'
+#' @return A sparse counts matrix
+#'
+#' @importFrom SeuratObject LayerData
+#'
+#' @export
+counts_matrix_from_assay <- function(assay) {
+  # Support both Assay and Assay5 classes
+  # The same as using the helper `assay$counts` but less indirection.
+  # With Assay5 it is important to use this method as oppossed to grabbing
+  # the data from `assay@layers` directly as that matrix won't contain the dimnames
+  SeuratObject::LayerData(assay, "counts")
```

Comment:
Per the above suggestion about keeping Seurat dependency unversioned, you could probably...


```
if (unlist(packageVersion("Seurat"))[1] > 5) {
  SeuratObject::LayerData(assay, "counts")
} else {
  assay@counts
}
```

Although I am not sure how that would interact with `NAMESPACE`

Reply:
Maybe something like this:

```
if (packageVersion("Seurat") >= 5) {
  SeuratObject::LayerData(assay, "counts")
} else {
  if (is(assay, 'Assay5')) {
    stop("Cannot get count matrix: Please upgrade to Seurat > 5 to support dataset")
  }

  assay@counts
}
```

It looks like it is possible to have a Seurat object with a mix of assay versions.

Reply:
The only way they get into the `else if` is if they download a Seurat 5 dataset while on Seurat < 5, right? I think that is a fine condition to `stop` on. Make sure to unlist the return of `packageVersion`, though (it is a list with each entry being a corresponding x/y/z)

Reply:
that is my understanding